### PR TITLE
[Needs testing] Added random (time seeded) init

### PR DESCRIPTION
**Pull request by theishshah**
Issue #51 feature should working, needs to be added to test script

### CODE REVIEW

**Comment by thomasdfischer** [comment]
My bad. By random, I meant we should initialize all the values to random values instead of default values.

For example

```
$ cat my_data.json
{
    "color": "red",
    "size": 20
}
$ cat my_data.json | schema infer | schema init
{
    "color": "",
    "size": 0,
}
$ # the output above has default values like the empty string and 0
$ # it might be useful to init with random values instead of default values
$ cat my_data.json | schema infer | schema init --random
{
    "color": "J3Kns24oIJOSo839d",
    "size": -10823.218943,
}
```

Sending normal JSON but with random values might be useful when testing APIs

**Comment by theishshah** [comment]
Working on being able to properly run, currently the test hangs or fails on 'init_json_string' depending on the randomly generated value, I suspect this is an issue with the generate random string function

File: jsonsch/init.go [hunk 1]
```diff
@@ -120,7 +128,125 @@ func appendTuple(tup tuple, dstSlice []interface{}, key string) ([]interface{},
...
+	default:
+		return nil, fmt.Errorf("element of array at key '%v' has unrecognized 'type' field '%v'", key, tup.Type)
+	}
+	return append(dstSlice, elem), nil
+}
+
+func generateRandomString(length int) string {
+	rand.Seed(time.Now().Unix())
+
+	var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
```
Comment (thomasdfischer) [inline jsonsch/init.go]: letters := "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"

That way `letters` will be a `string`, which signifies it's immutable. `[]rune` implies mutability.

File: jsonsch/init.go [hunk 2]
```diff
@@ -120,7 +128,125 @@ func appendTuple(tup tuple, dstSlice []interface{}, key string) ([]interface{},
...
+
+	randomString := make([]rune, length%2147483647)
```
Comment (thomasdfischer) [inline jsonsch/init.go]: I think we should change 2147483647 to a smaller number. We can hard-code it as like 64 or 128 for now and then make it into a flag later on.

File: jsonsch/init.go [hunk 3]
```diff
@@ -21,8 +23,14 @@ func InitSchema(s Schema, doPopLists bool) (map[string]interface{}, error) {
 			return nil, err
 		}
 
-		if err := storeTuple(tup, data, key, doPopLists); err != nil {
-			return nil, err
+		if doRandom {
+			if err := storeRandomTuple(tup, data, key, doPopLists); err != nil {
```
Comment (thomasdfischer) [inline jsonsch/init.go]: this is great

File: cmd/init.go [hunk 4]
```diff
@@ -52,6 +52,7 @@ func init() {
 	initCmd.Flags().BoolVarP(&initCfg.DoXml, "xml", "", false, "initialize as XML")
 	initCmd.Flags().BoolVarP(&initCfg.DoProtobuf, "protobuf", "", false, "initialize as protocol buffer")
 	initCmd.Flags().BoolVarP(&initCfg.DoGraphQL, "graphql", "", false, "initialize as GraphQL instance")
+	initCmd.Flags().BoolVarP(&initCfg.DoRandom, "random", "", true, "initialize with random values")
```
Comment (thomasdfischer) [inline cmd/init.go]: need to change this to default to `false` before merging

**Pull request closed without merging**